Adding shortest-paths documentation (#8187)
networkx/networkx: doc/reference/algorithms/shortest_paths.rst

 >>> print(weighted_path)
 ['A', 'C', 'E', 'D', 'F']
 >>> # Compute shortest path length from A to F ignoring weights
 >>> unweighted_path = nx.shortest_path(G, source="A", target="F")
 >>> print(unweighted_path)
 ['A', 'B', 'D', 'F']

Algorithms
----------

Depending on the type of graph and problem, different algorithms can perform
better than others. The table below summarizes the algorithms NetworkX supports
and their typical time complexities. Here, :math:`V` is the number of nodes and
:math:`E` is the number of edges in the graph.

+----------------------+-------------+--------------------+---------------------------+------------------------------------+
| Algorithm            | Weighted    | Query Type         | Time Complexity           | Recommended for                    |
|                      | Supported?  |                    |                           |                                    |
+======================+=============+====================+===========================+====================================+
| Breadth–First Search | Unweighted  | Single-source,     | :math:`O(V + E)`          | Fastest choice for unweighted      |

> |                      | Only        | Single-pair        |                           | graphs; shortest path in hops      |

Reviewer: ```suggestion
| (BFS)                | Only        | Single-pair        |                           | graphs; shortest path in hops      |
```

| ``source``       | ``target``       | Query Type                                  |
+==================+==================+=============================================+
| ``None``         | ``None``         | All pairs shortest paths                    |
+------------------+------------------+---------------------------------------------+
| specified        | ``None``         | From source to all reachable nodes          |
+------------------+------------------+---------------------------------------------+
| ``None``         | specified        | From all nodes that can reach the target    |
+------------------+------------------+---------------------------------------------+
| specified        | specified        | Single source–target shortest path          |
+------------------+------------------+---------------------------------------------+

By default, the simplified interface uses Breadth–First Search for unweighted
graphs and Dijkstra's algorithm for weighted graphs (``weight`` parameter is
not ``None``). In the special case of single-pair queries (both ``source`` and
``target`` specified), bidirectional variants of these algorithms are used.

> This default behavior can be overridden by specifying the ``method`` parameter.
> Algorithms Dijkstra, Bellman-Ford and Breadth–First Search (BFS) are supported.
> If an unsupported method is provided, NetworkX will raise an error.

Reviewer: ```suggestion
The default algorithm can be overridden to select Bellman-Ford by specifying
the ``method`` parameter to be "bellman-ford". Algorithms other than Dijkstra,
Bellman-Ford and Breadth–First Search are not supported in the simplified interface.
```

For the case of single-pair queries, bidirectional variants of Dijkstra and BFS
are used when those methods are specified.

> 

Reviewer: This is already included in the last sentence of the paragraph about default behavior. 
Do you want it both places or is one enough?

Author: What I was thinking here is that, even if you specify, for example, "dijkstra" you get bi-directional Dijstra. So let me remove it from the first parrargaph and leave it on the second!

Shortest Paths
==============

The shortest path problem involves finding a path between two nodes in a graph
such that the total distance is minimized. In unweighted graphs this means
finding the path with the fewest number of edges. In weighted graphs it is the
path with minimum sum of weights associated to the path edges.

This problem definition applies for both undirected and directed graphs. In
undirected graphs, edges can be traversed in any direction. In directed graphs
edges can only be followed in their defined direction, and the path must respect
those constraints. For more details on graph types and their properties, see
:ref:`graph types <classes>`.

> NetworkX provides a unified interface for shortest paths weighted and unweighed,

Reviewer 2: ```suggestion
NetworkX provides a unified interface for shortest paths weighted and unweighted,
```

directed and undirected. Other variants of the shortest path problem such as all
pairs of shortest paths are also supported.

> To specify graph is weighted, user must provide a weight attribute name by using

Reviewer 2: ```suggestion
To specify that graph is weighted, user must provide a weight attribute name by using
```

the ``weight`` parameter. This can be a string that corresponds to an edge
attribute or a function that returns the weight of an edge. If no weight is
specified, the graph is treated as unweighted. In the case ``weight`` is
specified, but the edge does not have the specified attribute, the edge is
treated as having a weight of :math:`1`.

 >>> import networkx as nx
 >>>
 >>> # Create an undirected weighted graph
 >>> G = nx.Graph()
 >>> G.add_edge("A", "B", weight=4)
 >>> G.add_edge("A", "C", weight=2)
 >>> G.add_edge("B", "C", weight=5)
 >>> G.add_edge("B", "D", weight=10)
 >>> G.add_edge("C", "E", weight=3)
 >>> G.add_edge("E", "D", weight=4)
 >>> G.add_edge("D", "F", weight=11)
 >>>
 >>> # Compute shortest path from A to F
 >>> path = nx.shortest_path(G, source="A", target="F", weight="weight")
 >>> print(path)

>  ['A', 'C', 'E', 'D', 'F']

Reviewer 2: We could consider expanding the example to demonstrate the unweighted result too!

Author: Good idea!

Algorithm Selection
-------------------

Depending on the type of graph and problem, different algorithms can perform

> better than others. When using the simplified interface, NerworkX picks the

Reviewer 2: ```suggestion
better than others. When using the simplified interface, NetworkX picks the
```

> algorithm that suits the use case best. Selection is based on the type of graph

Reviewer 2: ```suggestion
algorithm that best suits the use-case. The selection is based on the type of graph
```
